js/function.js

```diff
@@ -0,0 +1,445 @@
+    }
+    else {
+        showOnScreen.value = result;
+    }
+}
+function factorial(x) {                  //Factorial function
+    if (x == 0 || x == 1) {
+        return 1;
+    }
+    //if number is negative
+    else if (x < 0) {
+        x = Math.abs(x);
+        return x * factorial(x - 1);
+    }
+    // if number is positive
+    else {
+        return x * factorial(x - 1);
+    }
+}
+function Operator() {                //Dot operator function
+    var last = showOnScreen.value[showOnScreen.value.length-1];
+    console.log(last);
+    if(last != "."){
+        if (showOnScreen.value == "") {
+            showOnScreen.value = "0.";
+        }
+        else {
+            showOnScreen.value += ".";
+        }
+    }
+}
+function BackSpace() {               //Backspace function
+    if (showOnScreen.value != "") {
+        showOnScreen.value = showOnScreen.value.slice(0, -1);
+    }
+}
+function clickclr() {               //Clear the screen function for Display 1
+    showOnScreen.value = "";
+}
+function clickclr2() {              //Clear the screen function for Display 2
+    showOnScreen2.value = "";
+}
+function Display2() {                //Display the screen function for Display 2
+    let z = showOnScreen.value + "=";
+    showOnScreen2.value = z;
+}
+
+
+function clickequ() {               //On click Equal to function
+    if (showOnScreen.value.includes("^")) {
+        XYSquare();
+    }
+    if (showOnScreen.value.includes("Yroot")) {
+        XYroot();
+    }
+    if (showOnScreen.value.includes("baseY")) {
+        calculatelogBaseY();
+    }
+    var Display = eval(showOnScreen.value);
+    Display2();
+    showOnScreen.value = Display;
+}
+
+
+function trigonometry(x) {               //Trigonometry functions
+    switch (x) {
+        case "sin":
+            showOnScreen.value = Math.sin(showOnScreen.value);
+            break;
+        case "cos":
+            showOnScreen.value = Math.cos(showOnScreen.value);
+            break;
+
+        case "tan":
+            showOnScreen.value = Math.tan(showOnScreen.value);
+            break;
+
+        case "sec":
+            showOnScreen.value = 1 / Math.cos(showOnScreen.value);
+            break;
+
+        case "cosec":
+            showOnScreen.value = 1 / Math.sin(showOnScreen.value);
+            break;
+
+        case "cot":
+            showOnScreen.value = 1 / Math.tan(showOnScreen.value);
+            break;
+    }
+}
+function Functions(x) {                  //Mathematical functions
+    switch (x) {
+        case "floor":
+            showOnScreen.value = Math.floor(showOnScreen.value);
+            break;
+        case "ceil":
+            showOnScreen.value = Math.ceil(showOnScreen.value);
+            break;
+
+        case "mod":
+            showOnScreen.value = Math.abs(showOnScreen.value);
+            break;
+    }
+}
+function Revert(x) {                 //Change the value by pressing '2nd' button
+    if (ButtonClicked % 2 == 0) {
```

Comment:
Can we optimize this function some other way?

Reply:
Method 1: while clicking on the button we are checking odd even condition. In other ways we can do counter-- which can display other values.

Method:2 we can make 2 functions while clicking, it can call one function which makes style display block and that function call other function (callback function) which makes style display none and will increase the flag. in 2nd method we have to use 2 function while the current scenario has one function for the one operation.